### ISSUE TITLE
[issue] Elementor Pro Slides background overlay issues

### ISSUE THREAD

**Issue by BennyRum** [elementor/elementor#9180]
## Description

In the latest version of Elementor Pro, using the background-overlay setting on the Slide element has some issues: One is that the overlay is z-indexed above the content, making it unable to click on the button. The second is that the overlay is confined within the bounds of .swiper-slide-inner, which is 66% width instead of the desired 100%. See screenshot at [Imgur](https://imgur.com/a/rboqcZd) or live at [Bier Garten Website](https://biergartenbaycity.com/)

## Steps to reproduce

1. Have the latest versions of Elementor and Elementor Pro installed on a default WordPress theme
2. Use the Slides element (Elementor Pro) within a stretched, full-width section
3. In a slide, add a background image and enable the background overlay setting.

## Isolating the problem

- [X] This bug happens with only Elementor plugin active (and Elementor Pro).
- [X] This bug happens with a default WordPress theme active.
- [X] I can reproduce this bug consistently using the steps above.

## Environment

<details>
<summary>System Info</summary>
```
					
== Server Environment ==
	Operating System: Linux
	Software: Apache
	MySQL version: MariaDB Server v5.6.44-86.0
	PHP Version: 7.0.33
	PHP Max Input Vars: 8000
	PHP Max Post Size: 2047M
	GD Installed: Yes
	ZIP Installed: Yes
	Write Permissions: All right
	Elementor Library: Connected

== WordPress Environment ==
	Version: 5.2.3
	Site URL: https://biergartenbaycity.com
	Home URL: https://biergartenbaycity.com
	WP Multisite: No
	Max Upload Size: 2 GB
	Memory limit: 2047M
	Permalink Structure: /%postname%/
	Language: en-US
	Timezone: America/Detroit
	Debug Mode: Inactive

== Theme ==
	Name: weblocal2018
	Version: 1.0
	Author: weblocal
	Child Theme: No

== User ==
	Role: administrator
	WP Profile lang: en_US
	User Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/77.0.3865.90 Safari/537.36

== Active Plugins ==
	AddToAny Share Buttons
		Version: 1.7.37
		Author: AddToAny

	Autoptimize
		Version: 2.5.1
		Author: Frank Goossens (futtta)

	Breadcrumb NavXT
		Version: 6.3.0
		Author: John Havlik

	Contact Form 7
		Version: 5.1.4
		Author: Takayuki Miyoshi

	Contact Form CFDB7
		Version: 1.2.4.7
		Author: Arshid

	Elementor
		Version: 2.7.3
		Author: Elementor.com

	Elementor Pro
		Version: 2.7.0
		Author: Elementor.com

	Honeypot for Contact Form 7
		Version: 1.14.1
		Author: Nocean

	List Pages Shortcode
		Version: 1.7.4
		Author: Ben Huson, Aaron Harp

	Loginizer
		Version: 1.4.2
		Author: Raj Kothari

	Meta Tag Manager
		Version: 2.1
		Author: Marcus Sykes

	PDF Embedder
		Version: 4.4
		Author: Lever Technology LLC

	Simple Universal Google Analytics
		Version: 1.0.4
		Author: naa986

	Slimstat Analytics
		Version: 4.8.7.3
		Author: Jason Crouse

	Smush
		Version: 3.2.4
		Author: WPMU DEV

	Testimonials Widget
		Version: 3.5.0
		Author: Axelerant

	W3 Total Cache
		Version: 0.10.1
		Author: Frederick Townes

	WEBL_CRM
		Version: 1.1
		Author: WebLocal Inc


== Must-Use Plugins ==
	Elementor Safe Mode
		Version: 1.0.0
		Author: Elementor.com


== Log<a href="/wp-admin/admin.php?page=elementor-system-info&elementor-clear-log=1&_wpnonce=f561756b96#elementor-clear-log" class="box-title-tool">Clear Log</a><span id="elementor-clear-log"></span> ==
	
Log: showing 20 of 412019-09-24 14:35:13 [info] Elementor Pro/Upgrades - _v_2_5_0_posts Start  
2019-09-24 14:35:13 [info] Elementor Pro/Upgrades - _v_2_5_0_posts Finished 
2019-09-24 14:35:13 [info] Elementor Pro/Upgrades - _v_2_5_0_portfolio Start  
2019-09-24 14:35:13 [info] Elementor Pro/Upgrades - _v_2_5_0_portfolio Finished 
2019-09-24 14:35:13 [info] Elementor Pro/Upgrades - _v_2_5_0_products Start  
2019-09-24 14:35:13 [info] Elementor Pro/Upgrades - _v_2_5_0_products Finished 
2019-09-24 14:35:13 [info] Elementor Pro/Upgrades - _v_2_5_0_sitemap Start  
2019-09-24 14:35:13 [info] Elementor Pro/Upgrades - _v_2_5_0_sitemap Finished 
2019-09-24 14:35:13 [info] Elementor Pro/Upgrades - _v_2_5_0_popup_border_radius Start  
2019-09-24 14:35:13 [info] Elementor Pro/Upgrades - _v_2_5_0_popup_border_radius Finished 
2019-09-24 14:35:13 [info] Elementor Pro/Upgrades - _v_2_5_4_posts Start  
2019-09-24 14:35:13 [info] Elementor Pro/Upgrades - _v_2_5_4_posts Finished 
2019-09-24 14:35:13 [info] Elementor Pro/Upgrades - _v_2_5_4_portfolio Start  
2019-09-24 14:35:13 [info] Elementor Pro/Upgrades - _v_2_5_4_portfolio Finished 
2019-09-24 14:35:13 [info] Elementor Pro/Upgrades - _v_2_5_4_products Start  
2019-09-24 14:35:13 [info] Elementor Pro/Upgrades - _v_2_5_4_products Finished 
2019-09-24 14:35:13 [info] Elementor Pro/Upgrades - _v_2_5_4_form Start  
2019-09-24 14:35:13 [info] Elementor Pro/Upgrades - _v_2_5_4_form Finished 
2019-09-24 14:35:13 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor Pro',
  'from' => '2.3.1',
  'to' => '2.7.0',
)]
2019-09-24 14:35:13 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor Pro',
  'from' => '2.3.1',
  'to' => '2.7.0',
)]

JS: showing 2 of 2JS: 2019-05-29 23:40:04 [error X 2][https://biergartenbaycity.com/wp-content/plugins/elementor/assets/js/editor.min.js?ver=2.5.16:2:58271] i.get is not a function 
JS: 2019-05-29 23:40:55 [error X 1][https://biergartenbaycity.com/wp-content/plugins/elementor/assets/js/editor.min.js?ver=2.5.16:2:83489] Cannot set property \'controls\' of undefined 



== Elements Usage ==
	
	wp-post
		button : 4
		heading : 6
		html : 1
		icon : 3
		icon-box : 1
		image-carousel : 1
		shortcode : 1
		slides : 1
		text-editor : 9
		video : 1

				
```
</details>

**Comment by brycewray** [comment]
Same here. In my case, an existing Slides widget was set to a vertical height of 0 (apparently) and, when I fixed that, found the background overlay issue you described. I ended up using the new anything-can-have-a-Slides-background feature to replace it, but would just as soon have avoided the problem — and am not looking forward to having to do this with other existing Slides widgets I maintain.

**Comment by cirkut** [comment]
I'm having the same issue. The issue is that the slider background overlay (`.elementor-background-overlay`) has a `z-index` value set to `0`, while the slide content (`.`) has no `z-index` value at all, which puts it below the layer order, causing this background overlay issue.

My resolution for the meantime is to set Custom CSS on the slider to
`selector .swiper-slide-contents { z-index: 0;  }`.

**Comment by shilo-ey** [comment]
Hi @BennyRum 

Thanks for reporting,

We will address this issue and update this thread once a patch is released.

**Comment by parulgarg2** [comment]
There is also another issue with background overlay. After update it is only covering 66% of the area. It was fixed after I rolled back to the previous version.

![image](https://user-images.githubusercontent.com/34934968/65580054-95091680-df96-11e9-9897-ed4c05b418c7.png)

**Comment by tcco-ca** [comment]
Same is happening to us, same process (above):
Here's the URL `https://www.thcommercial.ca/`

**Comment by hackerfabian** [comment]
Hi at all,
had the same issue but today I saw you published an update which fixes the overlay bug. But there's another bug since upgrading to 2.7.0 (2.7.1):
I had a text field on a slider with a width of 50% and a left inner padding of 15% as you can see on the screenshot:

<img width="1919" alt="Bildschirmfoto 2019-09-26 um 22 45 57" src="https://user-images.githubusercontent.com/10959616/65723803-6f236500-e0af-11e9-8137-2197083aa3e2.png">

But instead of being on the left side the textbox is nearly centered. The issue is caused because the element with the class `swiper-slide-inner` isn't full width like before. When I remove the `margin: auto` in `.elementor-slides .swiper-slide-inner` it will work like in the further versions.

Hope you can fix it soon. ✌️

**Comment by dominicq** [comment]
Appears this problem has been fixed, but now the background overlay no longer functions when coupled with the key burns effect?

**Comment by daveadams85** [comment]
@dominicq I also experience this issue - the overlay no longer functions at all with Ken Burns.

Additionally, the horizontal position selectors (both in the style tab, and the content tab > style override) don't work.

**Comment by douglascreate** [comment]
Yes, elementor has broken my slider page. When is a new plugin update getting pushed out. This needs to be fixed very soon.

**Comment by shilo-ey** [comment]
> Appears this problem has been fixed, but now the background overlay no longer functions when coupled with the key burns effect?

Thanks for reporting.

We will address this issue soon and release a patch.

Please follow this thread for updates.

Thanks!

> @dominicq I also experience this issue - the overlay no longer functions at all with Ken Burns.
> 
> Additionally, the horizontal position selectors (both in the style tab, and the content tab > style override) don't work.

I can see the issue only in "content tab > style override" area. Does the Style tab Horizontal Positioning not working for someone else?

**Comment by hackerfabian** [comment]
> > Appears this problem has been fixed, but now the background overlay no longer functions when coupled with the key burns effect?
> 
> Thanks for reporting.
> 
> We will address this issue soon and release a patch.
> 
> Please follow this thread for updates.
> 
> Thanks!
> 
> > @dominicq I also experience this issue - the overlay no longer functions at all with Ken Burns.
> > Additionally, the horizontal position selectors (both in the style tab, and the content tab > style override) don't work.
> 
> I can see the issue only in "content tab > style override" area. Does the Style tab Horizontal Positioning not working for someone else?

Horizontal position not working for me in style tab:
https://github.com/elementor/elementor/issues/9180#issuecomment-535679781

**Comment by astbarc** [comment]
Hi,

I'm also having the issue with the horizontal position. I have slides with the horizontal content position set to left but the text appears centered.

**Comment by josieKNT** [comment]
Havin the same issue, my slider doesn't view at all in back or frontend. Only way to see it at all is chrome inspect.

**Comment by perks98** [comment]
@shilo-ey is there an ETA on an update to the pro plugin to fix the Ken Burns overlay issue? Haven't been able to update the plugin for a week due to issues around the slider.

Thanks.

**Comment by unclerom** [comment]
Overlay and horizontal alignment issues here too

**Comment by shilo-ey** [comment]
This issue has been resolved in Elementor Pro v2.7.2

Feel free to update

Thanks!
